### PR TITLE
Use Guid constructor instead of Guid.Parse...

### PR DESCRIPTION
**Pull request by KevinH-MS**
...for dynamic analysis instrumentation.  Guid.Parse is not available when
targeting .NET 2.0/3.5.

### CODE REVIEW

**Comment by tmat** [comment]
👍

**Comment by KevinH-MS** [comment]
FYI @JohnHamby @dotnet/roslyn-compiler 
/CC @dotnet/testimpact

**Comment by KevinH-MS** [comment]
Tagging @MattGertz for ask mode approval...

The scenario here is a solution that contains some projects targeting .NET 2.0 or 3.5.  Currently, building with dynamic analysis instrumentation turned on will fail for these projects, because we were inadvertently relying on a helper that was only available in .NET 4.0+.  Testing of real-world solutions on preview5 uncovered this issue.

**Comment by JohnHamby** [comment]
This looks good to me. Sorry for accidently introducing a later version dependency.

**Comment by MattGertz** [comment]
Approved.

File: src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb [hunk 1]
```diff
@@ -84,14 +84,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
 
-            Dim guidParse As MethodSymbol = factory.WellKnownMember(Of MethodSymbol)(WellKnownMember.System_Guid__Parse)
-            If guidParse IsNot Nothing Then
+            Dim guidConstructor As MethodSymbol = factory.WellKnownMember(Of MethodSymbol)(WellKnownMember.System_Guid__ctor)
+            If guidConstructor IsNot Nothing Then
                 body.Add(
                     factory.Assignment(
                        factory.ModuleVersionId(isLValue:=True),
-                       factory.Call(Nothing, guidParse, ImmutableArray.Create(factory.ModuleVersionIdString()))))
+                       factory.[New](guidConstructor, factory.ModuleVersionIdString())))
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb]: This change (on the VB side) seems to affect no tests. Can you make sure a VB test checks the IL for generated instrumentation?

**Comment by jcouv** [comment]
LGTM with a VB test suggestion.

File: src/Compilers/VisualBasic/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.vb [hunk 2]
```diff
@@ -1622,7 +1622,7 @@ End Class
 
             Dim diagnostics As ImmutableArray(Of Diagnostic) = CreateCompilation(source).GetEmitDiagnostics(EmitOptions.Default.WithInstrumentationKinds(ImmutableArray.Create(InstrumentationKind.TestCoverage)))
             For Each Diagnostic As Diagnostic In diagnostics
-                If Diagnostic.Code = ERRID.ERR_MissingRuntimeHelper AndAlso Diagnostic.Arguments(0).Equals("System.Guid.Parse") Then
+                If Diagnostic.Code = ERRID.ERR_MissingRuntimeHelper AndAlso Diagnostic.Arguments(0).Equals("System.Guid..ctor") Then
```
Comment (KevinH-MS) [inline src/Compilers/VisualBasic/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.vb]: Here is the VB test that changed.

File: src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb (hunk 1 above)
Comment (KevinH-MS) [inline src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb]: There is one VB test that changed (noted below), but it just checks the error behavior if the required Guid ctor doesn't exist.  I will add a test that baselines the IL.

Comment (KevinH-MS) [inline src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb]: Added test scenario for VB...

**Pull request merged**